time/rotate3.jl

```diff
@@ -0,0 +1,52 @@
+# rotate3.jl
+# time 3D image rotation using @thread and foreach
+
+using Revise
+using SPECTrecon
+using BenchmarkTools
+using Plots; default(markerstrokecolor=:auto)
+
+nx, nz = 128, 100
+T = Float32
+image3 = randn(T, nx,nx,nz)
+out1 = similar(image3)
+out2 = similar(image3)
+
+#plans = plan_rotate(nx; T, nthread = 99) # warns
+plans = plan_rotate(nx; T) # default nthread = Threads.nthreads())
+
+θ = π/6
+imrotate!(out1, image3, θ, plans, 17) # foreach
+imrotate!(out2, image3, θ, plans, :thread)
+@assert out1 == out2
+
+#@btime imrotate!($out1, $image3, $θ, $plans, 20) # 5.9 ms (888 allocations: 63.53 KiB)
+#@btime imrotate!($out1, $image3, $θ, $plans) # 6.1 ms (802 allocations: 56.77 KiB)
+#@btime imrotate!($out2, $image3, $θ, $plans, :thread) # 5.8 ms (41 allocations: 3.17 KiB)
```

Comment:
`foreach` is allocating?
I remembered `imrotate!` allocates zero memory using `Threads.@threads()`.